src/doc_table.c

```diff
@@ -9,24 +9,76 @@
 #include "rmalloc.h"
 #include "spec.h"
 
+#define MAX_SIZE 1000000
```

Comment:
`DOCTABLE_MAX_SIZE`

Reply:
I think it would be good to make this adjustable via an environment variable (and a constructor attribute) so that we can test what happens if we go over this limit witout actually exceeding the default limit.

Reply:
Made it a constructor variable (for unit tests)
But I do not see any advantage of make it adjustable via an environment variable, are you thinking of compiling a specific build for testing, not sure how good is it.

Reply:
Maybe configurable when the module starts? what do you think?